internal/tests/cluster/session_cleanup_test.go

```diff
@@ -363,6 +398,88 @@ func (s *testSession) TestNoConnectionsLeft(t *testing.T) {
 	require.Zero(t, s.connectionsLeft)
 }
 
+// ExpectConnectionStateOnWorker waits until all connections in a
+// session have transitioned to a particular state on the worker.
+func (s *testSession) ExpectConnectionStateOnWorker(
+	ctx context.Context,
+	t *testing.T,
+	tw *worker.TestWorker,
+	expectState session.ConnectionStatus,
+) {
+	t.Helper()
+	require := require.New(t)
+	assert := assert.New(t)
+
+	ctx, cancel := context.WithTimeout(ctx, expectConnectionStateOnWorkerTimeout)
+	defer cancel()
+
+	// This is just for initialization of the actual state set.
+	const sessionStatusUnknown session.ConnectionStatus = "unknown"
+
+	connIds := s.testWorkerConnectionIds(t, tw)
+	// Make a set of states, 1 per connection
+	actualStates := make(map[string]session.ConnectionStatus, len(connIds))
+	for _, id := range connIds {
+		actualStates[id] = sessionStatusUnknown
+	}
+
+	// Make expect set for comparison
+	expectStates := make(map[string]session.ConnectionStatus, len(connIds))
+	for _, id := range connIds {
+		expectStates[id] = expectState
+	}
+
+	for {
+		if ctx.Err() != nil {
+			break
+		}
+
+		for _, conn := range s.testWorkerConnectionInfo(t, tw) {
+			actualStates[conn.Id] = session.ConnectionStatusFromProtoVal(conn.Status)
+		}
+
+		if reflect.DeepEqual(expectStates, actualStates) {
```

Comment:
Not important to change here, but we tend to use go-cmp for this as it prints out much easier to understand information, like the exact differences.

Reply:
Yeah, this is not for assertion's sake (would have used testify if that was the case). This is just to check for equality.